examples/twittermap/web/public/javascripts/map/heatmap.js

```diff
@@ -1,28 +1,119 @@
 angular.module('cloudberry.map')
   .controller('heatMapCtrl', function($scope, $rootScope, $window, $http, $compile, cloudberry, leafletData, cloudberryConfig, Cache) {
     function setHeatMapStyle() {
+      $scope.setStyles({
+        initStyle: {
+          weight: 0.5,
+          fillOpacity: 0,
+          color: 'white'
+        },
+        stateStyle: {
+          fillColor: '#f7f7f7',
+          weight: 0.5,
+          opacity: 1,
+          color: '#92d1e1',
+          fillOpacity: 0
+        },
+        stateUpperStyle: {
+          fillColor: '#f7f7f7',
+          weight: 0.5,
+          opacity: 1,
+          color: '#92d1e1',
+          fillOpacity: 0
+        },
+        countyStyle: {
+          fillColor: '#f7f7f7',
+          weight: 0.5,
+          opacity: 1,
+          color: '#92d1e1',
+          fillOpacity: 0
+        },
+        countyUpperStyle: {
+          fillColor: '#f7f7f7',
+          weight: 0.5,
+          opacity: 1,
+          color: '#92d1e1',
+          fillOpacity: 0
+        },
+        cityStyle: {
+          fillColor: '#f7f7f7',
+          weight: 0.5,
+          opacity: 1,
+          color: '#92d1e1',
+          fillOpacity: 0
+        },
+        hoverStyle: {
+          weight: 0.7,
+          color: '#666',
+          fillOpacity: 0
+        },
+        colors: [ '#ffffff', '#92d1e1', '#4393c3', '#2166ac', '#f4a582', '#d6604d', '#b2182b'],
+        sentimentColors: ['#ff0000', '#C0C0C0', '#00ff00']
+      });
     }
     
     function cleanHeatMap() {
+      if ($scope.heatMapLayer){
+        $scope.map.removeLayer($scope.heatMapLayer);
+        $scope.heatMapLayer = null;
+      }
     }
     
     function setInfoControlHeatMap() {
-      //TODO For HeatMap use later.
+      function onEachFeature(feature, layer) {
+        layer.on({
+          click: $scope.zoomToFeature
+        });
+      }
+
+      $scope.loadGeoJsonFiles(onEachFeature);
+      
+      $scope.resetZoomFunction(onEachFeature);
+      $scope.resetDragFunction(onEachFeature);
+
+      if (!$scope.heat){
+        $scope.heat = L.heatLayer([]);
+      }
     }
     
     function drawHeatMap(result) {
+      function setHeatMapPoints(points) {
+        $scope.heat.setLatLngs(points);
+        $scope.heat.redraw();
+      }
+      
+      if (!$scope.heatMapLayer){
+        $scope.heatMapLayer = $scope.heat;
+        $scope.map.addLayer($scope.heatMapLayer);
+      }
+        
+      var points = [];
+      for (var i = 0; i < result.length; i++) {
+        if (result[i].hasOwnProperty('coordinate')){
+          points.push([result[i].coordinate[1], result[i].coordinate[0]]);
+        }
+        else {
+          points.push([$scope.rangeRandom(result[i].id, result[i]["place.bounding_box"][0][1], result[i]["place.bounding_box"][1][1]), $scope.rangeRandom(result[i].id + 79, result[i]["place.bounding_box"][0][0], result[i]["place.bounding_box"][1][0])]);
```

Comment:
Why adding 79?

Reply:
It is just a magic number. We use tweet id as a random seed to generate longitude and latitude separately. If we use the same tweet id for the two values, they will always be the same and the resulting points will all be on the same orthogonal line. I will add a comment about it.